### PR TITLE
Use range-based loops in HFST transducer.

### PR DESCRIPTION
**Pull request by AmbientLighter**

### CODE REVIEW

File: libhfst/src/HfstTransducer.cc [hunk 1]
```diff
@@ -262,11 +262,10 @@ void HfstTransducer::insert_to_alphabet(const std::string &symbol)
...
 void HfstTransducer::insert_to_alphabet(const std::set<std::string> &symbols)
 {
-    for (std::set<std::string>::const_iterator it = symbols.begin();
-         it != symbols.end(); it++)
+    for (const auto & symbol : symbols)
       {
-        HfstTokenizer::check_utf8_correctness(*it);
-        if (*it == "")
+        HfstTokenizer::check_utf8_correctness(symbol);
+        if (symbol == "")
```
Comment (TinoDidriksen) [inline libhfst/src/HfstTransducer.cc:268]: That should be `if (symbol.empty())` - not your fault, but if you're going through things anyway...

File: libhfst/src/HfstTransducer.cc [hunk 2]
```diff
@@ -923,14 +914,12 @@ HfstTransducer::HfstTransducer(const StringPairVector & spv,
...
-    for (StringPairVector::const_iterator it = spv.begin();
-     it != spv.end(); it++)
+
+    for (const auto& it : spv)
       {
-    if (it->first == "" || it->second == "")
-      HFST_THROW_MESSAGE
-        (EmptyStringException,
-         "HfstTransducer(const StringPairVector&, ImplementationType)");
+        if (it.first == "" || it.second == "")
```
Comment (TinoDidriksen) [inline libhfst/src/HfstTransducer.cc:920]: Ditto .empty() x2

File: libhfst/src/HfstTransducer.cc [hunk 3]
```diff
@@ -3113,13 +3068,13 @@ void encode_flag_diacritics(HfstTransducer &fst)
...
   // copy alphabet, encode all flags
   StringSet alpha = basic_fst.get_alphabet();
-  for (StringSet::const_iterator it = alpha.begin(); it != alpha.end(); it++)
+  for (const auto & it : alpha)
     {
-      if (it->size() > 4)
+      if (it.size() > 4)
         {
-          if ((it->at(0) == '%') && (it->at(it->size()-1) == '%'))
+          if ((it.at(0) == '%') && (it.at(it.size()-1) == '%'))
```
Comment (TinoDidriksen) [inline libhfst/src/HfstTransducer.cc:3075]: Hm, there's `.at()` in HFST? Any use of `.at()` should be questioned. Usually `op[]` is fine, and it certainly is here.

File: libhfst/src/HfstTransducer.cc (hunk 1 above)
Comment (AmbientLighter) [reply libhfst/src/HfstTransducer.cc]: Ok, fixed.

File: libhfst/src/HfstTransducer.cc (hunk 2 above)
Comment (AmbientLighter) [reply libhfst/src/HfstTransducer.cc]: Ok, fixed.

File: libhfst/src/HfstTransducer.cc (hunk 3 above)
Comment (AmbientLighter) [reply libhfst/src/HfstTransducer.cc]: Ok, fixed.

**Pull request merged**